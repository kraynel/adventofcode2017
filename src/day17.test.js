const { getNextValue } = require("./day17");

describe("getNextValue", () => {
  it("get the next value", () => {
    expect(getNextValue(1, 3)).toBe(0);
    expect(getNextValue(2, 3)).toBe(0);
    expect(getNextValue(3, 3)).toBe(1);
    expect(getNextValue(4, 3)).toBe(1);
    expect(getNextValue(5, 3)).toBe(3);
    expect(getNextValue(6, 3)).toBe(2);
    expect(getNextValue(7, 3)).toBe(1);
    expect(getNextValue(8, 3)).toBe(2);
    expect(getNextValue(9, 3)).toBe(6);
    expect(getNextValue(10, 3)).toBe(5);
    expect(getNextValue(11, 3)).toBe(4);
    expect(getNextValue(2018, 3)).toBe(638);
  });

  it("should get the value", () => {
    expect(getNextValue(2018, 359)).toMatchSnapshot();
  });
});
