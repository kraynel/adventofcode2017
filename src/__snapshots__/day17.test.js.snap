// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNextValue should get the value 1`] = `1506`;

exports[`getValueAfterZero should get the value 1`] = `39479736`;
